src/map/mob.cpp

```diff
@@ -4015,8 +4015,21 @@ int mob_clone_spawn(struct map_session_data *sd, int16 m, int16 x, int16 y, cons
 	sd->fd = 0;
 
 	//Go Backwards to give better priority to advanced skills.
-	for (i=0,j = MAX_SKILL_TREE-1;j>=0 && i< MAX_MOBSKILL ;j--) {
-		uint16 skill_id = skill_tree[pc_class2idx(sd->status.class_)][j].skill_id;
+	std::shared_ptr<s_skill_tree> tree = skill_tree_db.find(sd->status.class_);
+
+	if (tree == nullptr || tree->skills.empty())
+		return 0;
```

Comment:
This "return" cannot be done like this.
It is okay if a clonse does not have skills.
Also if you return here, the "fd" of the cloned player will be lost.